internal/docstore/driver/driver.go

```diff
@@ -90,11 +90,18 @@ type ActionListError []struct {
 // A Query defines a query operation to find documents within a collection based
 // on a set of requirements.
 type Query struct {
-	FieldPaths [][]string       // the selected field paths
-	Filters    []Filter         // a list of filter for the query
-	Limit      int              // the number of result in one query request
-	StartAt    interface{}      // define the starting point of the query
-	Iter       DocumentIterator // the iterator for Get action
+	// FieldPaths contain a list of field paths user selects to return in the query
+	// results. The returned documents should only have these fields populated.
+	FieldPaths [][]string
+
+	// Filters contain a list of filters for the query. If there are more than one
+	// filter, they should be combined with AND.
+	Filters []Filter
+
+	// Limit sets the maximum number of results returned by one query request. It
+	// is always non-negative. When it <= 0, the driver implementation should
```

Comment:
The semantics of this are unclear to me. It feels like the comment is about retrieving a page of result, but the API is not paginated. The `Limit` seems like it should be specifically about capping the maximum number of times you can call `Next` in the return iterator, right? And the driver must always return all of the matching results, but if there are more than `Limit`, it can stop.

It also seems like `0` should mean "no limit", not a reasonable default. If the driver is using a paginated implementation, it can decide how big the pages it gets are.